core/src/main/java/org/elasticsearch/search/builder/SearchSourceBuilder.java

```diff
@@ -126,9 +126,9 @@ public static HighlightBuilder highlight() {
 
     private QueryBuilder postQueryBuilder;
 
-    private int from = -1;
+    private Integer from = null;
```

Comment:
I think that this change that introduces boxing should be reverted. This means that we need a value to represent unset, and -1 is the idiom within Elasticsearch for that.